version: "3.8"

networks:
  backend:
    external: true

services:
  postgres:
    image: postgres:16
    restart: always
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
      - TYPEBOT_DB_USER=${TYPEBOT_DB_USER}
      - TYPEBOT_DB_PASSWORD=${TYPEBOT_DB_PASSWORD}
      - N8N_DB_USER=${N8N_DB_USER}
      - N8N_DB_PASSWORD=${N8N_DB_PASSWORD}
    volumes:
      - /home/jsancosta/init-data.sh:/docker-entrypoint-initdb.d/init-data.sh
      - /home/jsancosta/data/db_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d postgres" ]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - backend